src/Schedule.h

```diff
@@ -117,48 +163,61 @@ class LoopLevel {
     // @}
 
     /** Construct an undefined LoopLevel. Calling any method on an undefined
-     * LoopLevel (other than defined() or operator==) will assert. */
-    LoopLevel() = default;
+     * LoopLevel (other than set()) will assert. */
+    EXPORT LoopLevel();
 
-    /** Return true iff the LoopLevel is defined. */
+    /** Return true iff the LoopLevel is defined. (Only LoopLevels created
+     * with the default ctor are undefined.) */
     EXPORT bool defined() const;
 
-    /** Return the Func name. Asserts if the LoopLevel is_root() or is_inline(). */
+    /** Mutate our contents to match the contents of 'other'. This can only be
+     * done to an unlocked LoopLevel. */
+    EXPORT void set(const LoopLevel &other);
+
+    /** Lock this LoopLevel. */
+    EXPORT LoopLevel &lock();
+
+    /** Return the Func name. Asserts if the LoopLevel is_root() or is_inline() or !defined(). */
     EXPORT std::string func() const;
 
-    /** Return the VarOrRVar. Asserts if the LoopLevel is_root() or is_inline(). */
+    /** Return the VarOrRVar. Asserts if the LoopLevel is_root() or is_inline() or !defined(). */
     EXPORT VarOrRVar var() const;
 
     /** inlined is a special LoopLevel value that implies
-     * that a function should be inlined away. */
+     * that a function should be inlined away. Asserts if !defined(). */
     EXPORT static LoopLevel inlined();
```

Comment:
I think this should be named `inline`... or is that not legal C++? If so, I think `is_inline` should be renamed to `is_inlined` to mirror this.

Reply:
Naming a method "inline" is definitely not gonna fly. The name predates this change but I'm fine with changing it.